### ISSUE TITLE
[Feat]: Add coachmark popover

### ISSUE THREAD

**Issue by adixon-adobe** [adobe/spectrum-web-components#2613]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

coachmark

### Description of the requested feature

It'd be great to get the coachmark popover present in [spectrum css](https://opensource.adobe.com/spectrum-css/coachmark.html) implemented as a web component (likely with placement options for associating with the actual coachmark)

### Implementation notes or ideas

Haven't thought through it much, but on first thought it'd be cool to allow both standalone usage, and inline placement within the `<sp-coachmark>` element. We have use cases for both the tutorial steps variant, and the single message variant.

**Comment by nickschaap** [comment]
Since coachmark popovers are mainly used in sequence with multiple different targets I was thinking a possible implementation for this feature could look something like the following:

```
<sp-coachmark-provider cancel-label="Skip tour" primary-label="Next" group="tour" active-step="1">

...

<sp-coachmark-target for="tour">
    <sp-button>Hello world!</sp-button>
    <sp-coachmark-popover slot="coachmark" title="Learn about the world!" content="We live in interesting times..." step="1" img-src="https://picsum.photos/200/300">
    </sp-coachmark-popover>
</sp-coachmark-target>

<sp-coachmark-target for="tour">
    <sp-button>Hello universe!</sp-button>
    <sp-coachmark-popover slot="coachmark" title="Learn about the universe!" content="Things are pretty dark and spacey here." step="2" img-src="https://picsum.photos/200/300">
    </sp-coachmark-popover>
</sp-coachmark-target>

...

</sp-coachmark-provider>
```

Notes:
- This implementation would likely build off of the `@lit-labs/context` package.
- The labels are controlled by the provider since they would stay consistent throughout the tour.

**Comment by adixon-adobe** [comment]
Apologies for not checking this out earlier. Overall this looks really close to what we need.

It'd be nice for the content to be a slot, since I could see needing to add something beyond just text there.

Similar for the footer (but I could see that being an enhancement beyond this work). E.g. on Lightroom we credit the tutorial author in the footer of each step to the left of the buttons.

**Comment by Westbrook** [comment]
Fixed by #3666
